### PR TITLE
[O11y][Oracle WebLogic] Add dimension fields for threadpool datastream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

This PR adds dimension fields for threadpool datastream to support TSDB enablement.

**Issue**: In the ThreadPool datastream, the data collection is performed by Jolokia input and there are different MBeans used to perform data collection. Because of different MBeans, multiple events are generated at the same time. And I don't have found any possible unique identifier field in fields.yml file for threadpool datastream. Hence the data was getting dropped while performing the data collection because of getting multiple events at the same time.

**Possible Solution**: To overcome the above issue, I have introduced a unique identifier called `oracle_weblogic.threadpool.mbean` that represents the value of the doc that is collected from the particular MBean and makes it dimension true.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

- Relates #7512

## Screenshot

Refer: https://github.com/elastic/integrations/issues/7512#:~:text=keyword%22%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%22ignore_above%22%3A%201024%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%7D-,ThreadPool,-%22oracle_weblogic%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22properties%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22threadpool

### CODE REVIEW

File: packages/oracle_weblogic/data_stream/threadpool/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -14,6 +14,14 @@ processors:
       target_field: oracle_weblogic
       ignore_missing: true
       ignore_failure: true
+  - rename:
+      field: oracle_weblogic.mbean
+      target_field: oracle_weblogic.threadpool.mbean
+      ignore_missing: true
+  - rename:
+      field: oracle_weblogic.threadpool.temp.objectName
```
Comment (agithomas) [inline packages/oracle_weblogic/data_stream/threadpool/elasticsearch/ingest_pipeline/default.yml:22]: Why this change to have a new field having name `oracle_weblogic.threadpool.temp.objectName` ?

Comment (harnish-elastic) [reply packages/oracle_weblogic/data_stream/threadpool/elasticsearch/ingest_pipeline/default.yml:22]: @agithomas, In this data stream, there are two different mbeans used. Now if the mbean has `*` (com.bea:ServerRuntime=`*`,Name=ThreadPoolRuntime,Type=ThreadPoolRuntime), we are getting mbean names in the response through the Jolokia input by creating the `oracle_weblogic.mbean` field. But if the mbean doesn't have `*` (`java.lang:type=Threading`), we are not getting the `oracle_weblogic.mbean` through the Jolokia input.

The issue here is the data is being ingested for both mbean within the identical timestamps. Also, there are no unique identifiers in the fields.yml file. Hence there was getting a data loss while enabling TSDB enablement. 

To overcome this issue, I have introduced one field in the [handlebars](https://github.com/elastic/integrations/pull/7517/files/3cbcd1a65f4b236de1346e0be8c22437a4d05586#diff-22bb743ba526fda39daa89308eaa000a78f75ea42c5dcb0e4dffd397a8513b32R44-R51). This field will provide the MBean name for the mbean who doesn't have `*` (`java.lang:type=Threading`). and give a field name `threadpool.temp`.

Now the Jolokia input will provide a mbean name by creating the `oracle_weblogic.threadpool.temp.objectName` field. We have just renamed this field to `oracle_weblogic.threadpool.mbean`.

Comment (agithomas) [reply packages/oracle_weblogic/data_stream/threadpool/elasticsearch/ingest_pipeline/default.yml:22]: I hope, it is validated that both rename is done to the same field `oracle_weblogic.threadpool.mbean` and there is no dataloss

Comment (harnish-elastic) [reply packages/oracle_weblogic/data_stream/threadpool/elasticsearch/ingest_pipeline/default.yml:22]: Yes @agithomas, it is validated that both renames are done to the same field the oracle_weblogic.threadpool.mbean and there is no data loss.

**Review by agithomas — APPROVED** [review]
LGTM! 
Please complete peer review additionally.

**Review by milan-elastic — APPROVED** [review]
LGTM!
